scheduler/launchd_test.go

```diff
@@ -0,0 +1,109 @@
+package scheduler
+
+import (
+	"os"
+	"testing"
+
+	"github.com/fluidkeys/fluidkeys/assert"
+)
+
+var ld = launchd{}
+
+func TestLaunchdEnable(t *testing.T) {
+	t.Run("enable successfully if file is present", func(t *testing.T) {
+		mockFileHelper := mockFileFunctions{
+			OsStatReturnError:          nil,
+			IoutilWriteFileReturnError: nil,
+		}
+		launchdEnabled, err := ld.enable(&mockFileHelper, "fake.plist")
+		assert.NoError(t, err)
+
+		assert.Equal(t, true, launchdEnabled)
+	})
+
+	t.Run("load successfully if file is missing", func(t *testing.T) {
+		mockFileHelper := mockFileFunctions{
+			OsStatReturnError:          os.ErrNotExist,
+			IoutilWriteFileReturnError: nil,
+		}
+		launchdEnabled, err := ld.enable(&mockFileHelper, "fake.plist")
+		assert.NoError(t, err)
+
+		assert.Equal(t, true, launchdEnabled)
+
+		t.Run("writes out default launchd content with correct mode", func(t *testing.T) {
+			assert.Equal(t, LaunchdFileContents, string(mockFileHelper.IoutilWriteFileGotData))
+			assert.Equal(t, os.FileMode(0644), mockFileHelper.IoutilWriteFileGotMode)
```

Comment:
You're explicitly testing that it should be written out with `0644` access permissions ... why those permissions?

Reply:
I looked at what the file permissions of all the other `.plist` files I have in that directory are for other applications - they are `-rw-r--r--`

Reply:
Good man!

Even so, I suggest to use `600`, I don't see why we want any other users being able to read the file.

Reply:
Roger.